docs/resources/vrf.md

```diff
@@ -7,81 +10,438 @@ description: |-
-
-
-Note: In the APIC GUI,a VRF (fvCtx) was called a "Context"or "PrivateNetwork."
-
-## Attribute Reference ##
-
-The only attribute that this resource exports is the `id`, which is set to the
-Dn of the VRF.
+All examples for the VRF resource can be found in the [examples](https://github.com/CiscoDevNet/terraform-provider-aci/tree/master/examples/resources/aci_vrf) folder.
+
+## Schema ##
+
+### Required ###
+
+* `parent_dn` - (string) The distinguished name (DN) of the parent object, possible resources:
+  - [aci_tenant](https://registry.terraform.io/providers/CiscoDevNet/aci/latest/docs/resources/tenant) ([fvTenant](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvTenant/overview))
+* `name` (name) - (string) The name of the VRF object.
+
+### Read-Only ###
+
+* `id` - (string) The distinguished name (DN) of the VRF object.
+
+### Optional ###
+
+* `annotation` (annotation) - (string) The annotation of the VRF object.
+  - Default: `orchestrator:terraform`
+* `bd_enforcement` (bdEnforcedEnable) - (string) The Bridge Domain (BD) enforcement status of the VRF. BD enforcement will restrict a host from sending ping requests to subnets outside the Bridge Domain (BD). The feature blocks ICMP, TCP and UDP traffic between BDs.
+  - Default: `false`
+  - Valid Values: `no`, `yes`.
+* `description` (descr) - (string) The description of the VRF object.
+* `ip_data_plane_learning` (ipDataPlaneLearning) - (string) The IP data plane status of the VRF object. The process known as IP dataplane learning involves acquiring the endpoint's IPv4 or IPv6 address through dataplane routing of traffic from the endpoint.
+  - Default: `enabled`
+  - Valid Values: `disabled`, `enabled`.
+* `name_alias` (nameAlias) - (string) The name alias of the VRF object.
+* `owner_key` (ownerKey) - (string) The key for enabling clients to own their data for entity correlation.
+* `owner_tag` (ownerTag) - (string) A tag for enabling clients to add their own data. For example, to indicate who created this object.
+* `policy_control_enforcement_direction` (pcEnfDir) - (string) The policy control enforcement direction of the VRF object. VRF ingress policy enforcement involves ACL filtering on the leaf switch where the endpoint is located, reducing the policy CAM usage on the border leaf switch by confining the filtering to 'compute' leaf switches. VRF egress policy enforcement extends ACL filtering to the border leaf switch, increasing its policy CAM usage. The border leaf switch handles filtering for traffic from L3Out to EPG after the endpoint is learned, unless the endpoint-to-destination class mapping is unknown, in which case the compute leaf switch handles the filtering.
+  - Default: `ingress`
+  - Valid Values: `egress`, `ingress`.
+* `policy_control_enforcement_preference` (pcEnfPref) - (string) The policy control enforcement preference of the VRF object.
```

Comment:
This does not look like a preference but more of a state and the explanation string is not clear.

Reply:
the name is like this in the UI, do you prefer deviating from UI
![Screenshot 2025-04-11 at 11 41 12](https://github.com/user-attachments/assets/261a4012-7abe-477c-8539-5627af596eb4)

Reply:
mode